### PR TITLE
Escape IOMD files (fixes #1416)

### LINKED ISSUE #1416: iomd is not HTML-escaped, allowing arbitrary code execution
re #382 (which this supercedes) -- now that we're not focused on the single runnable file/bundle user story, it's really not necessary to patch the IOMD into that script tag. We should just send the IOMD down as a string in a IOMD object or script, or whatever mechanism we're using to send over eg userData. This will solve #382 and also maybe simplify our start up code a teeny bit.

### PR DESCRIPTION
**Pull request by ZiyaoWei**
Sorry for the turnaround time!

This fixes #1416 by:

1) change IOMD content in the notebook template to be escaped instead of marked safe;
2) unescape the rendered content on load in JS.

Tests have been added to reflect the changes. This is kind of use visible but is a bit of an edge case so will update CHANGELOG.md if you think it's warranted.

LMK if there's a better approach!

### Pull Request checklist

- [ ] **Documentation**: If this feature has or requires documentation, the relevant docs have been updated.
- [x] **Changelog**: This PR updates the [changelog](https://github.com/iodide-project/iodide/blob/master/CHANGELOG.md) with any user-visible changes.
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not

### CODE REVIEW

File: src/editor/initialization/__tests__/handle-initial-iomd.test.js [hunk 1]
```diff
@@ -71,3 +71,37 @@ blah blah
...
   });
 });
+
+describe("handleInitialIomd unescapes #iomd elt content", () => {
+  const iomdString = `
+
+%% js
+blah &lt;&gt;&#39;&quot;&amp; blah
+
+    `;
```
[the remark this answers is not in the document]
Comment (ZiyaoWei) [reply src/editor/initialization/__tests__/handle-initial-iomd.test.js]: Sure! FYI - this is copypasta from `handleInitialIomd is correct when #iomd elt has content`, so should I remove those as well?

[the remark this answers is not in the document]
Comment (wlach) [reply src/editor/initialization/__tests__/handle-initial-iomd.test.js]: Hmm, I would suggest leaving those as-is for now. The newlines feel a little more awkward in python than javascript, and making modifications to the jest tests feels a little out of scope for this PR.

**Comment by wlach** [comment]
BTW, yes, this change seems worth mentioning in the changelog! Please add an entry.

**Comment by ZiyaoWei** [comment]
Turns out I completely forgot about the preview window on the right and the script in `md` and `html` sections are not showing up there. Will look into this - should that be a separate PR (apologies for random questions - still trying to understand Iodide's dev process)?

**Comment by wlach** [comment]
Definitely we would want this to work end-to-end before merging so please go ahead and make additions to this PR.

**Comment by ZiyaoWei** [comment]
Thanks for the review @wlach! Don't have a specific goal in mind, just want to help build out the next great programming environment :-) Jokes aside, I would be happy to take a look at #2175 although since iodide is completely weekend and nights for me can't promise a timeline. I might also look into compiling some libraries with pyodide and write some notebooks and I'll let you know if I went into that rabbit hole.

**Comment by ZiyaoWei** [comment]
Also FWIW preview rendering seems to be working if I add `_.escape` to report-pane.jsx - just want to leave a trace here in case you want to revisit supporting that.
